src/arith_uint256.cpp

```diff
@@ -187,7 +187,7 @@ unsigned int base_uint<BITS>::bits() const
         {
             for (int bits = 31; bits > 0; bits--)
             {
-                if (pn[pos] & 1 << bits)
+                if (pn[pos] & 1U << bits)
                     return 32 * pos + bits + 1;
```

Comment:
I believe we should also pull in https://github.com/bitcoin/bitcoin/commit/b14db5abab405a708f0166293f1ea12222a6bf03

Reply:
Agreed, I miss your comment for some reason. Will do asap.